src/sentry/api/event_search.py

```diff
@@ -356,29 +369,42 @@ def is_percentage_key(self, key):
         return key in self.percentage_keys
 
     def visit_search(self, node, children):
-        return self.flatten(children)
+        return self.flatten(self.remove_space(children[1]))
+
+    def visit_boolean_term(self, node, children):
+        if not self.allow_boolean:
+            raise InvalidSearchQuery(
+                'Boolean statements containing "OR" or "AND" are not supported in this search'
+            )
+
+        children = self.flatten(self.remove_space(children))
+        return children[0].text.upper()
+
+    def visit_paran_term(self, node, children):
```

Comment:
`visit_paren_term`

Reply:
damn does this do nothing then lol

Reply:
anyway this is gonna go away :)

Reply:
irrelevant now that `paren_term` went away.